src/pages/index.js

```diff
@@ -25,28 +36,135 @@ const cardData = {
 const userInfo = new UserInfo({
   nameSelector: ".profile__title",
   jobSelector: ".profile__description",
+  avatarSelector: ".profile__image",
 });
 
+api.getUserInfo().then((userData) => {
+  //console.log("Setting avatar to:", userData.avatar);
+  //console.log("User Data from API:", userData);
+  userInfo.setUserInfo({
+    name: userData.name,
+    job: userData.about,
+    avatar: userData.avatar,
+  });
+});
+
+/* ------------------------------------------------- */
+/*                     Form Validation
+/* ------------------------------------------------- */
+const formValidators = {};
+
+const enableValidation = (settings) => {
+  const formList = Array.from(document.querySelectorAll(settings.formSelector));
+  formList.forEach((formElement) => {
+    const validator = new FormValidator(settings, formElement);
+    const formName = formElement.getAttribute("name");
+
+    formValidators[formName] = validator;
+    validator.enableValidation();
+  });
+};
+
+enableValidation(settings);
+
 // /* ------------------------------------------------- */
-// /*                 Profile Edit Form
+// /*                 Profile Avatar
 /* ------------------------------------------------- */
+const handleAvatarFormSubmit = (formData) => {
+  const newAvatarUrl = formData.avatarUrl;
+
+  return api
+    .updateAvatar(newAvatarUrl)
+    .then((userData) => {
+      console.log("Avatar updated on server:", userData.avatar);
+      userInfo.setUserAvatar(userData.avatar);
+      profileEditAvatar.close();
+    })
+    .catch((error) => {
+      console.error("Error updating avatar:", error);
+    });
+};
+
+const profileEditAvatar = new PopupWithForm(
+  "#update-avatar-modal",
+  handleAvatarFormSubmit,
+  (formValues) => {
+    profileEditAvatar.renderLoading(true);
+    const avatarUrl = formValues.avatar;
+    api
+      .updateAvatar(avatarUrl)
+      .then((userData) => {
+        console.log("Avatar URL:", userData.avatar);
+        userInfo.setUserAvatar(userData.avatar);
+        profileEditAvatar.close();
+        validator.resetValidation();
+      })
+      .catch((err) => {
+        console.error(`Error Submitting Form: ${err}`);
+      })
+      .finally(() => {
+        console.log("Resetting button text");
+        profileEditAvatar.renderLoading(false);
+      });
+  }
+);
+
+const profileEditAvatarButton = document.querySelector(".profile__edit-icon");
+
+const formName = profileEditAvatar.getForm().getAttribute("name");
+//console.log("Form Element:", formName);
+
+const validator = formValidators[formName];
+if (validator) {
+  validator.disableButton();
+} else {
+  console.error(`No validator found for form "${formName}"`);
```

Comment:
You don't need it because it's just a validator of the `avatar` form`.  The name is too neutral


It should be named `avatarValidator`, but you don't need it, actually


```js

const avatarValidator = formValidators[formName];

```